tools/buildsteps/windows/prepare-env.bat

```diff
@@ -13,11 +13,14 @@ IF EXIST %WORKSPACE%\project\Win32BuildSetup\BUILD_WIN32 rmdir %WORKSPACE%\proje
 rem also clean 'build' dir used to build ffmpeg as git clean has trouble to remove some times
 IF EXIST %WORKSPACE%\project\BuildDependencies\build rmdir %WORKSPACE%\project\BuildDependencies\build /S /Q
 
+rem daemonized gpg-agent blocks mingw from being cleaned with git
+TASKKILL /IM "gpg-agent.exe" /F >nul 2>&1
+
 rem we assume git in path as this is a requirement
 rem git clean the untracked files and directories
 rem but keep the downloaded dependencies
 rem also keeps MSYS2 installation
-SET GIT_CLEAN_CMD=git clean -xffd -e "project/BuildDependencies/downloads" -e "project/BuildDependencies/downloads2" -e "project/BuildDependencies/mingwlibs" -e "project/BuildDependencies/msys64" -e "project/BuildDependencies/tools"
+SET GIT_CLEAN_CMD=git clean -xffd -e "project/BuildDependencies/downloads" -e "project/BuildDependencies/downloads2" -e "project/BuildDependencies/mingwlibs" -e "project/BuildDependencies/msys64" -e "project/BuildDependencies/tools" -e "cmake/addons/build/download/msys2-base-x86_64-20231026.tar.xz"
```

Comment:
Does this need to use a regex so it doesn't have to be updated whenever msys2 changes version?

Reply:
It looks like git clean doesn't accept regexes. There's not much cost in not changing this if the msys2 version changes, as it only takes ~20s to download. We only update msys2 every few years so I don't think it's a big issue.

Reply:
why would you want to exclude a clean of this file explicitly anyway?

Reply:
* wildcard seems to work fine locally.

```
c:\Users\brent\source\repos\xbmc>notepad msys-base-x86_64-20231026.tar.xz

c:\Users\brent\source\repos\xbmc>notepad msys2-base-x86_64-20231026.tar.xz

c:\Users\brent\source\repos\xbmc>dir *.tar.xz
 Volume in drive C has no label.
 Volume Serial Number is C07A-15D6

 Directory of c:\Users\brent\source\repos\xbmc

19/01/2024  05:06 PM                 0 msys-base-x86_64-20231026.tar.xz
19/01/2024  05:03 PM                 0 msys2-base-x86_64-20231026.tar.xz
               2 File(s)              0 bytes
               0 Dir(s)  140,702,515,200 bytes free

c:\Users\brent\source\repos\xbmc>git clean -xffd -e msys2-base-x86_64-*.tar.xz
Removing msys-base-x86_64-20231026.tar.xz

c:\Users\brent\source\repos\xbmc>dir *.tar.xz
 Volume in drive C has no label.
 Volume Serial Number is C07A-15D6

 Directory of c:\Users\brent\source\repos\xbmc

19/01/2024  05:03 PM                 0 msys2-base-x86_64-20231026.tar.xz
               1 File(s)              0 bytes
               0 Dir(s)  140,702,457,856 bytes free

c:\Users\brent\source\repos\xbmc>git --version
git version 2.35.1.windows.2
```

Reply:
Wildcard works fine.

Reply:
The file is needed by 160 game add-ons. I think it makes sense to cache a 57MB download on every run.

Reply:
I tried to cache the pacman packages too. I added the path `cmake\addons\build\mingw\src\mingw\var\cache\pacman\pkg` to excludes, but the state would get funky when it re-ran. It still downloads another 50MB of packages, it'd be nice to be able to cache those too.